### PR TITLE
Add more properties for `xor`

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Taken directly from `Idris2` - https://github.com/idris-lang/Idris2/blob/4fcb0fb4a796650f5df101526a923c0aa522d1ad/libs/base/Data/Bool/Xor.idr#L13-L64

The type checker is happy, but I am not sure if everything is right, because the `Idris` version of some properties look comparatively more complicated.

Also, please let me know if I should link the original `Idris` properties in the comments.

### CODE REVIEW

File: src/Data/Bool/Properties.agda [hunk 1]
```diff
@@ -627,20 +627,7 @@ true  <? _     = no  (λ())
...
-
-xor-∧-commutativeRing : CommutativeRing 0ℓ 0ℓ
-xor-∧-commutativeRing = ⊕-∧-commutativeRing
-  where
-  open BooleanAlgebraProperties ∨-∧-booleanAlgebra
-  open XorRing _xor_ xor-is-ok
-
-------------------------------------------------------------------------
--- Miscellaneous other properties
+-- Propertied of not
```
Comment (Saransh-cpp) [inline src/Data/Bool/Properties.agda:630]:
```suggestion
-- Properties of not
```

**Comment by gallais** [comment]
Dashed names typically only describe the LHS (e.g. `map-replicate`, not `map-replicate-replicate`) so:
`xor-same` rather than `xor-same-true`, `xor-true` rather than `xor-true-not`.

File: src/Data/Bool/Properties.agda [hunk 2]
```diff
@@ -660,6 +647,56 @@ not-¬ {false} refl ()
...
+
+xor-is-ok : ∀ x y → x xor y ≡ (x ∨ y) ∧ not (x ∧ y)
+xor-is-ok true  y = refl
+xor-is-ok false y = sym (∧-identityʳ _)
+
+xor-same : ∀ x → x xor x ≡ false
+xor-same false = refl
+xor-same true  = refl
+
+xor-false : ∀ x → false xor x ≡ x
```
Comment (MatthewDaggitt) [inline src/Data/Bool/Properties.agda:661]: In the same way we do for `and` and `or` can we use the pre-defined types in `Algebra.Definitions`  e.g. identity, comm etc, and change the names to match?

**Comment by JacquesCarette** [comment]
I'll work with @Saransh-cpp to get soem more stdlib-sounding names in this and the next batch of PRs. And also to use the stdlib types.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1985,10 +1985,18 @@ Other minor changes
...
-  ∧-conical : Conical true _∧_
+  ∧-conical  : Conical true _∧_
+  xor-same-false    : ∀ x → x xor x ≡ false
+  xor-false-neutral : ∀ x → false xor x ≡ x
+  xor-true-not      : ∀ x → true xor x ≡ not x
+  xor-not-true      : ∀ x → x xor (not x) ≡ true
+  not-xor           : ∀ x y → not (x xor y) ≡ (not x) xor y
+  not-xor-cancel    : ∀ x y → (not x) xor (not y) ≡ x xor y
+  xor-commutative   : ∀ x y → x xor y ≡ y xor x
+  xor-associative   : ∀ x y z → x xor (y xor z) ≡ (x xor y) xor z
```
Comment (Saransh-cpp) [inline CHANGELOG.md:1999]: Will update once the property names and types and final.

Comment (MatthewDaggitt) [reply CHANGELOG.md:1999]: Thanks. Once the CHANGELOG is updated, this looks good to me.

Comment (Saransh-cpp) [reply CHANGELOG.md]: Thanks for the review, updated!

**Pull request merged**